keps/sig-multicluster/2149-clusterid/README.md

```diff
@@ -0,0 +1,756 @@
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+The new multi-cluster services API (see [KEP-1645](https://github.com/kubernetes/enhancements/tree/master/keps/sig-multicluster/1645-multi-cluster-services-api)) 
+expanded the ways clusters can communicate with each other and organized them into `ClusterSet`s, but as of now there is no way for a cluster to be uniquely identified 
+in a Kubernetes-native way. This document by SIG-Multicluster proposes a standard 
+for how cluster IDs should be stored and managed, based on concrete use cases 
+discussed and observed in `ClusterSet` deployments. While existing implementations may not currently or plan to abide by this standard, future expansions to the Multi-Cluster API will be designed on top of this standard and existing MCS API implementations are encouraged to adopt it.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+That there must be some way to identify individual clusters in a multi-cluster 
+deployment has felt like a given to SIG-Multicluster; it has been discussed in
+a broad sense previously ([see this doc](https://docs.google.com/document/d/1F__vEKeI41P7PPUCMM9PVPYY34pyrvQI5rbTJVnS5c4/edit?usp=sharing), and was scoped 
+down in response to actual observed use cases in the latest community discussion on
+which this KEP is based ([doc](https://docs.google.com/document/d/1S0u6xzP2gcJKPipA6tBNDNuid76nVKeGhTk7PrCIuQY/edit?usp=sharing)). The motivation
+of this KEP is to provide a flexible but useful baseline for clusterID that can
+work with the known use cases (see the User Stories section). 
+
+Existing implementations of the MCS API may have addressed the need for a cluster ID in their own ways, inconsistent with this current standard. It is the perspective of SIG-Multicluster that future additions to the MCS API will depend when necessary on the proposal laid out here, and existing implementations are encouraged to migrate any existing cluster ID assignment and storage mechanism to fit within the specifications of this KEP.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+* Propose a standard for how cluster identification metadata should be stored and 
```

Comment:
So this goals section needs a lot of updating.  From reading below it really sounds like you are trying to enable a set of different user stories, but they should really be called out up above.

Reply:
That's a fair point. I would expect to see something like:

- Establish reliable coordinates for determining clusterset membership and identity of a cluster within its cluster set
- Enable cluster-granularity DNS names for multicluster services
- Facilitate enrichment of log / event / metrics data with cluster id / set coordinates

Reply:
Stole your prose and added it to the goals section explicitly in 74f9290 :)